domain/src/main/java/me/fontys/semester4/postalcode/PCConverter.java

```diff
@@ -0,0 +1,93 @@
+package me.fontys.semester4.postalcode;
+
+import com.healthmarketscience.jackcess.CursorBuilder;
+import com.healthmarketscience.jackcess.Database;
+import com.healthmarketscience.jackcess.DatabaseBuilder;
+import com.healthmarketscience.jackcess.IndexCursor;
+import me.fontys.semester4.data.entity.PostalcodePart;
+import me.fontys.semester4.data.repository.PostalcodePartRepository;
+import me.fontys.semester4.interfaces.FileConverter;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component("pcConverter")
+public class PCConverter implements FileConverter
+{
+    private final static String PC_TABLE="POSTCODES";
+    private final static String PC_RANGE_TABLE="GEMEENTEN";
+    private final static String PC_ROW_PC="A13_POSTCODE";
+    private final static String PC_ROW_RANGE_TYPE="A13_REEKSIND";
+    private final static String PC_ROW_RANGE_START="A13_BREEKPUNT_VAN";
+    private final static String PC_ROW_RANGE_END="A13_BREEKPUNT_TEM";
+    public final List<PostalcodePart> postalcodeParts;
+
+
+    public PCConverter()
+    {
+        postalcodeParts = new ArrayList<>();
+    }
+
+
+
+    @Override
+    public void Convert(File importFile) throws IOException
+    {
+        Database anImport = new DatabaseBuilder(importFile)
+                .setFileFormat(Database.FileFormat.V2000)
+                .open();
+
+        // check tables
+        for (var table: new String[] {PC_TABLE,PC_RANGE_TABLE})
```

Comment:
var is technically allowed but wouldn't it be better to define the actual type?
`for (String table: new String[] {PC_TABLE,PC_RANGE_TABLE})`

Reply:
It might be a preference thing, but var, in a strongly typed object oriented language such as Java, is extremely confusing. Especially when it's used several times like here. I don't know what variable type `row` is. Same for `postalcode`, `even`, `fromrange`, `torange`. With all of these variables, I have to not only look at the definition, but also at the chained code that initialized the variable, just to figure out what the type is. It forces me to look back and forth, up and down, left and right, just to understand what this variable is, let alone understand what the code does.

Code is like a book; it should read smoothly. if you have to go back and re-read things you already read before, then you're not working efficiently. A book like Clean Code beautifully covers these kind of things.

Reply:
I will change it